### ISSUE TITLE
[BUG/ISSUE] Error in WASHOUT for wet deposition in 14.1.0

### ISSUE THREAD

**Issue by FeiYao-Edinburgh** [geoschem/geos-chem#1662]
### What institution are you from?
The University of Edinburgh

### Description of the problem
Hi GCST,

Similar but **NOT** identical to issues reported [here](https://github.com/geoschem/geos-chem/issues/501), I am coming across the following wet deposition issues on 05/01/2019. I am running the version 14.1.0 of the standard GEOS-Chem model from 20190101 with the Restart file downloaded from [here](http://ftp.as.harvard.edu/gcgrid/geos-chem/data/ExtData/GEOSCHEM_RESTARTS/GC_14.0.0/GEOSChem.Restart.fullchem.20190101_0000z.nc4) and with the MERRA-2 meteorology. I have not introduced any modifications to the model, except that I turn off offline grid-indpendent emissions to use their online alternatives.
```
WETDEP: ERROR at   40  67   1 for species    2 in area WASHOUT: at surface
 LS          :  T
 PDOWN       :    0.0000000000000000     
 QQ          :    0.0000000000000000     
 ALPHA       :    0.0000000000000000     
 ALPHA2      :    0.0000000000000000     
 RAINFRAC    :    0.0000000000000000     
 WASHFRAC    :    0.0000000000000000     
 MASS_WASH   :    0.0000000000000000     
 MASS_NOWASH :    0.0000000000000000     
 WETLOSS     :                        NaN
 GAINED      :    0.0000000000000000     
 LOST        :    0.0000000000000000     
 DSpc(NW,:)  :                        NaN   6.0358243778561746E-013   6.5871997362336500E-013   7.2710915872550685E-013   8.0185772698102585E-013   8.7883682997147595E-013   9.6396466805517407E-013   1.0574719517340253E-012   1.1617302070198606E-012   1.2976219851862141E-012   1.4347568254382824E-012   1.5772212240871896E-012   1.7071657565802178E-012   1.8443377617027378E-012   1.9982208320328261E-012   2.1567932874822908E-012   2.2591568422224307E-012   2.2208301198704935E-012   1.8475974519883714E-012   1.7716069173018996E-013   1.7714395985520433E-013   1.7633649101242403E-013   1.6668529114369137E-013   1.3548045738669223E-013   5.1061710020314286E-014   0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000     
 Spc(I,J,:N) :                        NaN   3.5108056785061143E-009   3.8363969256742307E-009   3.6615166033026556E-009   3.6780394914242783E-009   4.1462343168230006E-009   4.7319942271993657E-009   5.1961472823088513E-009   5.4030830279477525E-009   5.5736845790195336E-009   5.7139596145766606E-009   5.8629212873139874E-009   7.9742789235773213E-009   1.0334311421916619E-008   1.0816150360971255E-008   1.1168715310744298E-008   1.1534959217017146E-008   1.1809950282570185E-008   1.7969626885629474E-008   1.7430760762446019E-008   1.7477810715818748E-008   1.7967321756900857E-008   1.8683742574601477E-008   1.9309929368816065E-008   2.0262386892450682E-008   2.0489969814921647E-008   1.9961590106306151E-008   2.2859284477873924E-008   1.3161046290246557E-008   6.5857053651000387E-009   2.7535806161296159E-009   1.2708780077337107E-009   3.6557775667039418E-010   6.1984105316417057E-011   2.6665694620973736E-011   8.7599157145440813E-012   4.8009375158768866E-012   1.0086435318729046E-012   1.3493529625353547E-013   1.6403790023674963E-014   2.7417226109948757E-015   4.2031825835582592E-014   2.3778709382809943E-013   8.3223532851684382E-013   4.5695049346098890E-012   6.9911523125704209E-012   2.5076669266356582E-012
===============================================================================
===============================================================================
GEOS-Chem ERROR: Error encountered in wet deposition!
 -> at SAFETY (in module GeosCore/wetscav_mod.F90)
===============================================================================

===============================================================================
GEOS-Chem ERROR: Error encountered in "Safety"!
 -> at Do_Washout_at_Sfc (in module GeosCore/wetscav_mod.F90)
===============================================================================

===============================================================================
GEOS-Chem ERROR:
 -> at WetDep (in module GeosCore/wetscav_mod.F90)
===============================================================================

===============================================================================
GEOS-Chem ERROR: Error encountered in "Wetdep"!
 -> at Do_WetDep (in module GeosCore/wetscav_mod.F90)
===============================================================================

===============================================================================
GEOS-CHEM ERROR: Error encountered in "Do_WetDep"!
STOP at  -> at GEOS-Chem (in GeosCore/main.F90)
===============================================================================
     - CLEANUP: deallocating arrays now...

-------------------------------------------------------------------------------
HEMCO 3.6.0 FINISHED.
Warnings (level 1 or lower):    518
-------------------------------------------------------------------------------
```
As you can see, there are **three** variables, WETLOSS, DSpc, and Spc, that have got values of NaN, so I do not think it would be safe to force them back to zeros when that situation occurs, but I do think it is worth your attention.

Following your discussion, I am currently re-running the model but this time with a Restart [file](http://ftp.as.harvard.edu/gcgrid/geos-chem/10yr_benchmarks/14.0.0/GCClassic/Restarts/2019/GEOSChem.Restart.20190101_0000z.nc4) from the last 10-yr benchmark rather than the one shipped with the run directory. I will let you know if this will help resolve the issue. Btw, can you let me know what's the difference between the Restart file coming from the two differenent folders, i.e. [GEOSCHEM_RESTARTS](http://ftp.as.harvard.edu/gcgrid/data/ExtData/GEOSCHEM_RESTARTS/) and [Restarts](http://ftp.as.harvard.edu/gcgrid/geos-chem/10yr_benchmarks/14.0.0/GCClassic/Restarts/), as I have asked [here](https://github.com/geoschem/geos-chem/discussions/911)?

All the best,
Fei

### Description of troubleshooting performed
See above.

### GEOS-Chem version
GCClassic v14.1.0

### Description of modifications
N/A

### Log files
[GC.log](https://github.com/geoschem/geos-chem/files/10794928/GC.log)

### Software versions
- CMake version: cmake version 3.16.3
- Compilers (Intel or GNU, and version): GNU Fortran (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
- NetCDF version: netCDF 4.7.3, netCDF-Fortran 4.5.2

**Comment by lizziel** [comment]
Hi @FeiYao-Edinburgh,

>  Btw, can you let me know what's the difference between the Restart file coming from the two differenent folders, i.e. [GEOSCHEM_RESTARTS](http://ftp.as.harvard.edu/gcgrid/data/ExtData/GEOSCHEM_RESTARTS/) and [Restarts](http://ftp.as.harvard.edu/gcgrid/geos-chem/10yr_benchmarks/14.0.0/GCClassic/Restarts/), as I have asked [here](https://github.com/geoschem/geos-chem/discussions/911)?

The [Restarts](http://ftp.as.harvard.edu/gcgrid/geos-chem/10yr_benchmarks/14.0.0/GCClassic/Restarts/) folder contains restart files generated during the 10-year full chemistry benchmark of GEOS-Chem Classic 14.0.0. The [GEOSCHEM_RESTARTS](http://ftp.as.harvard.edu/gcgrid/data/ExtData/GEOSCHEM_RESTARTS/) folder contains all restart files that are linked to when creating run directories in different GEOS-Chem versions. In that folder, as you observed, the specialty simulation restart files are in directories indicating date and year created. In contrast, the full chemistry simulation restarts files are in directories indicating which version of GEOS-Chem was used to produce them. The restarts in `GC14.0.0` were generated running benchmark simulations of GEOS-Chem 14.0.0, with details included in the `README` within that folder.

**The GEOS-Chem restart file directory names do not reflect what GEOS-Chem version uses those restart files.** For example, GEOS-Chem 14.0.0 and 14.0.1 run directories use restart files in the `GC_13.0.0` folder. However, GEOS-Chem 14.0.2 uses the restarts generated during the 14.0.0 10-year benchmark. This is because the 14.0.0 10-year benchmark output was not yet available when 14.0.0 was released.

To know the source of a restart file that ships with the run directory you need to look at the symbolic link target, go to that directory, and read the `README`. For the case of 14.1.0 the restart files that come with the run directory were generated with the 10-year benchmark of 14.0.0. Did you diff the files before swapping out your restart file to check that they are not the same?

Regarding the wet deposition issue, did you do a continuous run from Jan 1, 2019 and then it crashed on May 1st? Did it crash in the first hour or at some other time during the day? If it was running starting back in January it seems unlikely to be a restart file issue. It would more likely be a corrupt input file. Have you run during this timespan with MERRA2 before? Are you restarting on May 1st or Jan 1st with a restart file you got from the 10-year benchmark directory? You can test if there's a corrupt input file by starting on May 1st.

Let us know how your testing goes.

**Comment by FeiYao-Edinburgh** [comment]
> In that folder, as you observed, the specialty simulation restart files are in directories indicating date and year created.

May I confirm that you mean subfolders like "v2022-10" under [GEOSCHEM_RESTARTS](http://ftp.as.harvard.edu/gcgrid/data/ExtData/GEOSCHEM_RESTARTS/)? Although I also see subfolders like "GC_14.0.0" under [GEOSCHEM_RESTARTS](http://ftp.as.harvard.edu/gcgrid/data/ExtData/GEOSCHEM_RESTARTS/).

> The restarts in `GC14.0.0` were generated running benchmark simulations of GEOS-Chem 14.0.0, with details included in the `README` within that folder.

This is great, answering my question on what's the difference between the fullchem Restart files under [GEOSCHEM_RESTARTS](http://ftp.as.harvard.edu/gcgrid/data/ExtData/GEOSCHEM_RESTARTS/) versus [Restarts](http://ftp.as.harvard.edu/gcgrid/geos-chem/10yr_benchmarks/14.0.0/GCClassic/Restarts/). The [README](http://ftp.as.harvard.edu/gcgrid/data/ExtData/GEOSCHEM_RESTARTS/GC_14.0.0/README) suggests that the restarts in `GC14.0.0` were generated running 10-year benchmark simulations of GEOS-Chem 14.0.0-rc.3 using MERRA-2 4x5 meteorology, so fullchem restart files are exactly identical under [GEOSCHEM_RESTARTS](http://ftp.as.harvard.edu/gcgrid/data/ExtData/GEOSCHEM_RESTARTS/) versus [Restarts](http://ftp.as.harvard.edu/gcgrid/geos-chem/10yr_benchmarks/14.0.0/GCClassic/Restarts/), with the latter containing a full list of restart files covering 2010-2020. I also confirm this by a `diff` command although I am not sure whether `diff` can work with netCDF files maybe I need a command like `ncdiff`?
```
$ diff /gcgrid/data/ExtData/GEOSCHEM_RESTARTS/GC_14.0.0/GEOSChem.Restart.fullchem.20190101_0000z.nc4 /gcgrid/geos-chem/10yr_benchmarks/14.0.0/GCClassic/Restarts/2019/GEOSChem.Restart.20190101_0000z.nc4
$ # no output
```
In this sense, you are right the wet deposition error is none of businees of restart files. As such my run that restarts from Jan 1 2019 using the standard restart file still crashes on **---> DATE: 2019/01/05  UTC: 00:40** (apologize for the inconsistent date formats I previously use). As you can see, it crashed on the first hour of the day. I have not run this timespan before, so I will follow your advice to check whether there are any corrupt files in my input data.
```
---> DATE: 2019/01/05  UTC: 00:40  X-HRS:     96.666664
     - DO_LINEAR_CHEM: Linearized chemistry at 2019/01/05 00:40
     - LINOZ_CHEM3: Doing LINOZ
===============================================================================
WETDEP: ERROR at   40  67   1 for species    2 in area WASHOUT: at surface
 LS          :  T
 PDOWN       :    0.0000000000000000     
 QQ          :    0.0000000000000000     
 ALPHA       :    0.0000000000000000     
 ALPHA2      :    0.0000000000000000     
 RAINFRAC    :    0.0000000000000000     
 WASHFRAC    :    0.0000000000000000     
 MASS_WASH   :    0.0000000000000000     
 MASS_NOWASH :    0.0000000000000000     
 WETLOSS     :                        NaN
 GAINED      :    0.0000000000000000     
 LOST        :    0.0000000000000000     
 DSpc(NW,:)  :                        NaN   6.0358243778561746E-013   6.5871997362336500E-013   7.2710915872550685E-013   8.0185772698102585E-013   8.7883682997147595E-013   9.6396466805517407E-013   1.0574719517340253E-012   1.1617302070198606E-012   1.2976219851862141E-012   1.4347568254382824E-012   1.5772212240871896E-012   1.7071657565802178E-012   1.8443377617027378E-012   1.9982208320328261E-012   2.1567932874822908E-012   2.2591568422224307E-012   2.2208301198704935E-012   1.8475974519883714E-012   1.7716069173018996E-013   1.7714395985520433E-013   1.7633649101242403E-013   1.6668529114369137E-013   1.3548045738669223E-013   5.1061710020314286E-014   0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000     
 Spc(I,J,:N) :                        NaN   3.5108056785061143E-009   3.8363969256742307E-009   3.6615166033026556E-009   3.6780394914242783E-009   4.1462343168230006E-009   4.7319942271993657E-009   5.1961472823088513E-009   5.4030830279477525E-009   5.5736845790195336E-009   5.7139596145766606E-009   5.8629212873139874E-009   7.9742789235773213E-009   1.0334311421916619E-008   1.0816150360971255E-008   1.1168715310744298E-008   1.1534959217017146E-008   1.1809950282570185E-008   1.7969626885629474E-008   1.7430760762446019E-008   1.7477810715818748E-008   1.7967321756900857E-008   1.8683742574601477E-008   1.9309929368816065E-008   2.0262386892450682E-008   2.0489969814921647E-008   1.9961590106306151E-008   2.2859284477873924E-008   1.3161046290246557E-008   6.5857053651000387E-009   2.7535806161296159E-009   1.2708780077337107E-009   3.6557775667039418E-010   6.1984105316417057E-011   2.6665694620973736E-011   8.7599157145440813E-012   4.8009375158768866E-012   1.0086435318729046E-012   1.3493529625353547E-013   1.6403790023674963E-014   2.7417226109948757E-015   4.2031825835582592E-014   2.3778709382809943E-013   8.3223532851684382E-013   4.5695049346098890E-012   6.9911523125704209E-012   2.5076669266356582E-012
===============================================================================
===============================================================================
GEOS-Chem ERROR: Error encountered in wet deposition!
 -> at SAFETY (in module GeosCore/wetscav_mod.F90)
===============================================================================

===============================================================================
GEOS-Chem ERROR: Error encountered in "Safety"!
 -> at Do_Washout_at_Sfc (in module GeosCore/wetscav_mod.F90)
===============================================================================

===============================================================================
GEOS-Chem ERROR:
 -> at WetDep (in module GeosCore/wetscav_mod.F90)
===============================================================================

===============================================================================
GEOS-Chem ERROR: Error encountered in "Wetdep"!
 -> at Do_WetDep (in module GeosCore/wetscav_mod.F90)
===============================================================================

===============================================================================
GEOS-CHEM ERROR: Error encountered in "Do_WetDep"!
STOP at  -> at GEOS-Chem (in GeosCore/main.F90)
===============================================================================
     - CLEANUP: deallocating arrays now...

-------------------------------------------------------------------------------
HEMCO 3.6.0 FINISHED.
Warnings (level 1 or lower):    518
-------------------------------------------------------------------------------
```


> To know the source of a restart file that ships with the run directory you need to look at the symbolic link target, go to that directory, and read the `README`.

The `README` file under a run directory only shows the following information:
```
# This run directory is for GEOS-Chem Classic.

## For instructions on setting up, compiling, and running GEOS-Chem Classic, see:

  - https://geos-chem.readthedocs.org

## For additional information about GEOS-Chem, see:

  - http://wiki.geos-chem.org

  - http://geos-chem.org

## For help with GEOS-Chem Classic, see:

  - https://geos-chem.readthedocs.io/en/latest/reference/SUPPORT.html

  - https://geos-chem.readthedocs.io/en/latest/reference/CONTRIBUTING.html
```
But I have found that `log.dryrun.unique` can tell which specific restart files that a specific GEOS-Chem version will use. Taking GEOS-Chem v14.1.0 as an illustration, its `log.dryrun.unique` contains the following info:
```
./Restarts/GEOSChem.Restart.20190101_0000z.nc4 --> /ExtData/GEOSCHEM_RESTARTS/GC_14.0.0/GEOSChem.Restart.fullchem.20190701_0000z.nc4
```
And as desribed by [README](http://ftp.as.harvard.edu/gcgrid/data/ExtData/GEOSCHEM_RESTARTS/GC_14.0.0/README), this restart file is obtained from the 10-year benchmark simulation for GEOS-Chem 14.0.0-rc.3 using MERRA-2 4x5 meteorology. Also note the wrong point of "GEOSChem.Restart.2019**01**01_0000z.nc4" to "GEOSChem.Restart.fullchem.2019**07**01_0000z.nc4", but that should only be a dryrun issue for which I have fixed manually.

I'll do more tests and update you in time.

**Comment by FeiYao-Edinburgh** [comment]
Hi GCST,

As a follow-up, the issue is not associated with the GEOS-Chem Restart files or the meteorological files on Jan 5 2019.

I re-compile `gcclassic` with `CMAKE_BUILD_TYPE` set to `Debug`. Then it becomes very weird that the job crashes immediately on 1 Jan 2019 likely due to errors in the `Debug` part of RRTMG source codes?
```
At line 1652 of file /exports/csce/datastore/geos/users/fyao2/Projects/climate-change-air-pollution-solar-energy/rundirs/201901/CodeDir/src/GEOS-Chem/GeosCore/rrtmg_rad_transfer_mod.F90
Fortran runtime error: Index '11' of dimension 3 of array 'state_diag%radaodwl1' above upper bound of 9
At line 1652 of file /exports/csce/datastore/geos/users/fyao2/Projects/climate-change-air-pollution-solar-energy/rundirs/201901/CodeDir/src/GEOS-Chem/GeosCore/rrtmg_rad_transfer_mod.F90
Fortran runtime error: Index '11' of dimension 3 of array 'state_diag%radaodwl1' above upper bound of 9
At line 1652 of file /exports/csce/datastore/geos/users/fyao2/Projects/climate-change-air-pollution-solar-energy/rundirs/201901/CodeDir/src/GEOS-Chem/GeosCore/rrtmg_rad_transfer_mod.F90
Fortran runtime error: Index '11' of dimension 3 of array 'state_diag%radaodwl1' above upper bound of 9
At line 1652 of file /exports/csce/datastore/geos/users/fyao2/Projects/climate-change-air-pollution-solar-energy/rundirs/201901/CodeDir/src/GEOS-Chem/GeosCore/rrtmg_rad_transfer_mod.F90
Fortran runtime error: Index '11' of dimension 3 of array 'state_diag%radaodwl1' above upper bound of 9
At line 1652 of file /exports/csce/datastore/geos/users/fyao2/Projects/climate-change-air-pollution-solar-energy/rundirs/201901/CodeDir/src/GEOS-Chem/GeosCore/rrtmg_rad_transfer_mod.F90
Fortran runtime error: Index '11' of dimension 3 of array 'state_diag%radaodwl1' above upper bound of 9
At line 1652 of file /exports/csce/datastore/geos/users/fyao2/Projects/climate-change-air-pollution-solar-energy/rundirs/201901/CodeDir/src/GEOS-Chem/GeosCore/rrtmg_rad_transfer_mod.F90
Fortran runtime error: Index '11' of dimension 3 of array 'state_diag%radaodwl1' above upper bound of 9
At line 1652 of file /exports/csce/datastore/geos/users/fyao2/Projects/climate-change-air-pollution-solar-energy/rundirs/201901/CodeDir/src/GEOS-Chem/GeosCore/rrtmg_rad_transfer_mod.F90
Fortran runtime error: Index '11' of dimension 3 of array 'state_diag%radaodwl1' above upper bound of 9
At line 1652 of file /exports/csce/datastore/geos/users/fyao2/Projects/climate-change-air-pollution-solar-energy/rundirs/201901/CodeDir/src/GEOS-Chem/GeosCore/rrtmg_rad_transfer_mod.F90
Fortran runtime error: Index '11' of dimension 3 of array 'state_diag%radaodwl1' above upper bound of 9
At line 1652 of file /exports/csce/datastore/geos/users/fyao2/Projects/climate-change-air-pollution-solar-energy/rundirs/201901/CodeDir/src/GEOS-Chem/GeosCore/rrtmg_rad_transfer_mod.F90
Fortran runtime error: Index '11' of dimension 3 of array 'state_diag%radaodwl1' above upper bound of 9
At line 1652 of file /exports/csce/datastore/geos/users/fyao2/Projects/climate-change-air-pollution-solar-energy/rundirs/201901/CodeDir/src/GEOS-Chem/GeosCore/rrtmg_rad_transfer_mod.F90
Fortran runtime error: Index '11' of dimension 3 of array 'state_diag%radaodwl1' above upper bound of 9
At line 1652 of file /exports/csce/datastore/geos/users/fyao2/Projects/climate-change-air-pollution-solar-energy/rundirs/201901/CodeDir/src/GEOS-Chem/GeosCore/rrtmg_rad_transfer_mod.F90
Fortran runtime error: Index '11' of dimension 3 of array 'state_diag%radaodwl1' above upper bound of 9
At line 1652 of file /exports/csce/datastore/geos/users/fyao2/Projects/climate-change-air-pollution-solar-energy/rundirs/201901/CodeDir/src/GEOS-Chem/GeosCore/rrtmg_rad_transfer_mod.F90
Fortran runtime error: Index '11' of dimension 3 of array 'state_diag%radaodwl1' above upper bound of 9
At line 1652 of file /exports/csce/datastore/geos/users/fyao2/Projects/climate-change-air-pollution-solar-energy/rundirs/201901/CodeDir/src/GEOS-Chem/GeosCore/rrtmg_rad_transfer_mod.F90
Fortran runtime error: Index '11' of dimension 3 of array 'state_diag%radaodwl1' above upper bound of 9
At line 1652 of file /exports/csce/datastore/geos/users/fyao2/Projects/climate-change-air-pollution-solar-energy/rundirs/201901/CodeDir/src/GEOS-Chem/GeosCore/rrtmg_rad_transfer_mod.F90
Fortran runtime error: Index '11' of dimension 3 of array 'state_diag%radaodwl1' above upper bound of 9

Error termination. Backtrace:

Error termination. Backtrace:
At line 1652 of file /exports/csce/datastore/geos/users/fyao2/Projects/climate-change-air-pollution-solar-energy/rundirs/201901/CodeDir/src/GEOS-Chem/GeosCore/rrtmg_rad_transfer_mod.F90
Fortran runtime error: Index '11' of dimension 3 of array 'state_diag%radaodwl1' above upper bound of 9

Error termination. Backtrace:

Error termination. Backtrace:

Error termination. Backtrace:

Error termination. Backtrace:
```

[GC.log](https://github.com/geoschem/geos-chem/files/10807001/GC.log)

To reproduce my errors, you can simply run `createRunDir.sh` of GEOS-Chem v14.1.0 with the following choices:
```
1. Full chemistry
8. RRTMG
1. MERRA-2 (Recommended)
2. 2.0  x 2.5
2. 47 (reduced)
```

Compile the model with `cmake -DRUNDIR=.. -DRRTMG=y ../CodeDir # Plus -DCMAKE_BUILD_TYPE=Debug`

I have only slightly modified `geoschem_config.yml` to restart from Jan 1 2019 using the corresponding GC restart [file](http://ftp.as.harvard.edu/gcgrid/data/ExtData/GEOSCHEM_RESTARTS/GC_14.0.0/GEOSChem.Restart.fullchem.20190101_0000z.nc4), `HEMCO_Config.rc` to use online rather than offline grid-independent emissions, and `HISTORY.rc` to output certain diagnostics every three hours. All commonly-updated configuration files and `species_database.yml` are attached below:

[HISTORY.rc.txt](https://github.com/geoschem/geos-chem/files/10806984/HISTORY.rc.txt)
[HEMCO_Diagn.rc.txt](https://github.com/geoschem/geos-chem/files/10806985/HEMCO_Diagn.rc.txt)
[HEMCO_Config.rc.gmao_metfields.txt](https://github.com/geoschem/geos-chem/files/10806986/HEMCO_Config.rc.gmao_metfields.txt)
[HEMCO_Config.rc.txt](https://github.com/geoschem/geos-chem/files/10806987/HEMCO_Config.rc.txt)
[geoschem_config.yml.txt](https://github.com/geoschem/geos-chem/files/10806988/geoschem_config.yml.txt)
[species_database.yml.txt](https://github.com/geoschem/geos-chem/files/10807007/species_database.yml.txt)

I am looking forward to your help. At the same time, I will try restarting from July 1 2019 to see if that's only the issue for Jan 1 2019.

All the best,
Fei

**Comment by FeiYao-Edinburgh** [comment]
I have switched `debug_printout` to `true` to collect more info. I attach below the GEOS-Chem log files if that may help resolve the issue:

[GC_turn_on_Debug_and_debug_printout.log](https://github.com/geoschem/geos-chem/files/10814315/GC_turn_on_Debug_and_debug_printout.log)
[GC_turn_on_Release_and_debug_printout.log](https://github.com/geoschem/geos-chem/files/10814316/GC_turn_on_Release_and_debug_printout.log)

**Comment by FeiYao-Edinburgh** [comment]
I have switched `debug_printout` to `true` to collect more info. I attach below the GEOS-Chem log files if that may help resolve the issue:

[GC_turn_on_Debug_and_debug_printout.log](https://github.com/geoschem/geos-chem/files/10814315/GC_turn_on_Debug_and_debug_printout.log)
[GC_turn_on_Release_and_debug_printout.log](https://github.com/geoschem/geos-chem/files/10814316/GC_turn_on_Release_and_debug_printout.log)

**Comment by lizziel** [comment]
Hi @FeiYao-Edinburgh, I am going to try to reproduce the error. Stay tuned.

**Comment by FeiYao-Edinburgh** [comment]
Hi @lizziel Thanks! Just let you know:

> I re-compile `gcclassic` with `CMAKE_BUILD_TYPE` set to `Debug`. Then it becomes very weird that the job crashes immediately on 1 Jan 2019 likely due to errors in the `Debug` part of RRTMG source codes?

This crash is associated with my modifications to `HISTORY.rc` in which I simply switch on/off certain collections and their frequnecy and duration. Using the default `HISTORY.rc`, the debug version of my `gcclassic` is currently running slowly but smoothly. However, that will NOT remove the wet deposition issue. Also, it is weird that slightly modifying `HISTORY.rc` can lead to such issues.

I also tried Luo_WetDep scheme but with no luck. The simulations still died at "2019/01/05 00:40".

What I am going to do next is to re-download all input from washu from scratch but I suspect if that will help. In any case I will update you in time.

**Comment by FeiYao-Edinburgh** [comment]
Agreed. I will do two more tests with `RadAODWL1` turned on and off to see if that's another issue of GEOS-Chem v14.1.0.

**Comment by lizziel** [comment]
Actually, if you could focus on the wetdep issue that would be great. I'm pretty sure the RadAODWL1 issue is a bug which I can test and fix on my end.

**Comment by lizziel** [comment]
See fix for the RRTMG diagnostics issue in https://github.com/geoschem/geos-chem/pull/1669/files. It will be included in 14.1.1.

**Comment by FeiYao-Edinburgh** [comment]
> Using the default `HISTORY.rc`, the debug version of my `gcclassic` is currently running slowly but smoothly. However, that will NOT remove the wet deposition issue.

My debug run is finished with the corresponding GC log file attached below. From this file I strongly feel that the wet deposition issue is associated with some errors in GC source codes that I really appreciate your help!
```
Program received signal SIGFPE: Floating-point exception - erroneous arithmetic operation.

Backtrace for this error:
#0  0x152b40210d21 in ???
#1  0x152b4020fef5 in ???
#2  0x152b3fe9308f in ???
	at /build/glibc-SzIz7B/glibc-2.31/signal/../sysdeps/unix/sysv/linux/x86_64/sigaction.c:0
#3  0x55db9c20f180 in __fullchem_sulfurchemfuncs_MOD_fullchem_sulfuraqchem
	at /exports/csce/datastore/geos/users/fyao2/Projects/climate-change-air-pollution-solar-energy/rundirs/201901/CodeDir/src/GEOS-Chem/KPP/fullchem/fullchem_SulfurChemFuncs.F90:207
#4  0x55db9ba5d066 in set_sulfur_chem_rates
	at /exports/csce/datastore/geos/users/fyao2/Projects/climate-change-air-pollution-solar-energy/rundirs/201901/CodeDir/src/GEOS-Chem/GeosCore/fullchem_mod.F90:1828
#5  0x55db9ba766f7 in __fullchem_mod_MOD_do_fullchem._omp_fn.1
	at /exports/csce/datastore/geos/users/fyao2/Projects/climate-change-air-pollution-solar-energy/rundirs/201901/CodeDir/src/GEOS-Chem/GeosCore/fullchem_mod.F90:1488
#6  0x152b4007778d in ???
#7  0x152b3fcbe608 in start_thread
	at /build/glibc-SzIz7B/glibc-2.31/nptl/pthread_create.c:477
#8  0x152b3ff6f132 in ???
	at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95
#9  0xffffffffffffffff in ???
```

[GC_turn_on_Debug_and_debug_printout_Updated.log](https://github.com/geoschem/geos-chem/files/10830855/GC_turn_on_Debug_and_debug_printout_Updated.log)

**Comment by FeiYao-Edinburgh** [comment]
> What I am going to do next is to re-download all input from washu from scratch but I suspect if that will help.

As expected the wet deposition issue has not gone even if I re-download all input files from washu from scratch. I'd be grateful if you could help with this annoying wet deposition issue at your earliest convenience with all the info I provided above. Also please let me know if there is anything else that I can do on my end.

**Comment by yantosca** [comment]
Thanks for writing @FeiYao-Edinburgh.  It may be a while before we can take a look at this.

In the meantime, have you tried to investigate the source of the error?  The traceback says that the 

```console
Backtrace for this error:
#0  0x152b40210d21 in ???
#1  0x152b4020fef5 in ???
#2  0x152b3fe9308f in ???
	at /build/glibc-SzIz7B/glibc-2.31/signal/../sysdeps/unix/sysv/linux/x86_64/sigaction.c:0
#3  0x55db9c20f180 in __fullchem_sulfurchemfuncs_MOD_fullchem_sulfuraqchem
	at /exports/csce/datastore/geos/users/fyao2/Projects/climate-change-air-pollution-solar-energy/rundirs/201901/CodeDir/src/GEOS-Chem/KPP/fullchem/fullchem_SulfurChemFuncs.F90:207
#4  0x55db9ba5d066 in set_sulfur_chem_rates
	at /exports/csce/datastore/geos/users/fyao2/Projects/climate-change-air-pollution-solar-energy/rundirs/201901/CodeDir/src/GEOS-Chem/GeosCore/fullchem_mod.F90:1828
#5  0x55db9ba766f7 in __fullchem_mod_MOD_do_fullchem._omp_fn.1
	at /exports/csce/datastore/geos/users/fyao2/Projects/climate-change-air-pollution-solar-energy/rundirs/201901/CodeDir/src/GEOS-Chem/GeosCore/fullchem_mod.F90:1488
#6  0x152b4007778d in ???
#7  0x152b3fcbe608 in start_thread
	at /build/glibc-SzIz7B/glibc-2.31/nptl/pthread_create.c:477
#8  0x152b3ff6f132 in ???
	at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95
#9  0xffffffffffffffff in ???
```

I would look at the fullchem_SulfurChemFuncs.F90 line 207 and see what that line is.  It could be a division by zero or an out-of-bounds error.

If you are using RRTMG, please also take a look at this bug fix, which will go into 14.1.1: https://github.com/geoschem/geos-chem/pull/1669

**Comment by FeiYao-Edinburgh** [comment]
Hi @yantosca I have looked at line 207 of `fullchem_SulfurChemFuncs.F90` (copied below) but honestly I have no idea what's happening there in the absence of an understanding of relevant variables.
```
206       ! Assume SO2 is limiting, so recompute rxn rate accordingly
207       K_MT(1) = kIIR1Ltd( C(ind_SO2), C(ind_SALAAL), k_ex ) / C(ind_O3)
```

Line 1828 of `fullchem_mod.F90` is calling `fullchem_SulfurAqChem` of `fullchem_SulfurChemFuncs.F90` so is an inheirted error.
```
1821       CALL fullchem_SulfurAqChem( I          = I,                           &
1822                                   J          = J,                           &
1823                                   L          = L,                           &
1824                                   Input_Opt  = Input_Opt,                   &
1825                                   State_Chm  = State_Chm,                   &
1826                                   State_Grid = State_Grid,                  &
1827                                   State_Met  = State_Met,                   &
1828                                   RC         = RC                          )
```

Line 1488 of `fullchem_mod.F90` is an evaluation whether to archive OH reactivity for which I don't know why it was also reported in the debug info.
```
1487       IF ( State_Diag%Archive_OHreactivity           .or.                   &
1488            State_Diag%Archive_SatDiagnOHreactivity ) THEN
```






> If you are using RRTMG, please also take a look at this bug fix, which will go into 14.1.1: #1669

Thanks for reminding. I have already included this bug fix myself. It is within expectation that this does not help resolve the wet deposition issue.



> It may be a while before we can take a look at this.

Thanks! I can kindly wait for your efforts.

**Comment by lizziel** [comment]
Hi @FeiYao-Edinburgh, here are a few things to try:

1. Make sure you are using a restart file that was spun up for the date of interest. I think you are but might have missed something. Don't use a July restart file for January. 
2. Check if a non-RRTMG run has the same problem.
3. Check if a 72-level run has the same problem.
4. Check if a 4x5 run has the same problem.
5. Turn on daily diagnostics if you haven't already, and check to see if anything is blowing up leading up to the problem date
6. Try turning off wet deposition and see if that fixes the problem with okay looking results, or if the model then crashes elsewhere. I suspect the problem is elsewhere and perhaps is only showing up first in wet deposition.

Regarding restart files, one of your earlier comments said that your January restart symbolic link pointed to July. Was it like that when you created the run directory? You can view the target of symbolic links by doing `file symlinkname` at the command line, where `symlinkname` is the name of the symbolic link.

**Comment by FeiYao-Edinburgh** [comment]
> * Make sure you are using a restart file that was spun up for the date of interest. I think you are but might have missed something. Don't use a July restart file for January.

Yes I can confirm that I am using the January restart file for simulations restarting at January. In addition, I have scheduled a simulation restarting at July using the July restart file. For your information, the July simulation run quite smoothly with a successful end.

> Regarding restart files, one of your earlier comments said that your January restart symbolic link pointed to July.

If my memory is correct, that's a very samll issue as related to the `log.dryrun.unique` generated by `./download_data.py log.dryrun --skip-download` if I haven't prepare the Jan restart file in `ExtData/GEOSCHEM_RESTARTS/` in advance. But for safety I always link restart files myself so that's NOT an issue.

To speed up our efforts, may I kindly confirm with you about the following particularly Option 2:
- Option 2: Do I need to re-compile the model with `-DRRTMG=n` **OR** do I simply need to deactivate `rrtmg_rad_transfer_model` in the `operations` menu in `geoschem_config.yml` and of course comment RRTMG collections in `HISTORY.rc`?
- Option 3: I simply assign 72 to `number_of_levels` in the `grid` menu in `geoschem_config.yml`.
- Option 4: I simply assign `4.0x5.0` to `resolution` in the `grid` menu in `geoschem_config.yml`.
- Option 5: I assign `00000001 000000` to `frequency` AND `duration` to ALL collections that are turned on.
- Option 6: I simply deactivate `wet_deposition` in the `operations` menu.

**Comment by FeiYao-Edinburgh** [comment]
Hi @lizziel I think you are right that I'd better to create new run directories for options 2 to 4. In particular:

For option 2, I have re-compiled the model without `-DRRTMG=y` so it is a standard fullchem.

While for options 3 and 4, I do not think I need to re-compile my model since grid menu changes can be captured during run time thanks to the flexgrid capability of geos-chem. So like running nested simulations, I definitely could copy and slightly modify my existing run directories to create run directories for options 3 and 4.

However, I also realize that it is currently VERY convenient to create run directories with `createRunDir.sh`, and using that I only find difference in `number_of_levels:` in `geoschem_config.yml` for option 3, while for option 4 I find difference in `resolution:` in `geoschem_config.yml`, `METDIR:` in `HEMCO_Config.rc` and `HEMCO_Config.rc.gmao_metfields`, and `Mass tuning factor     :` (as related to DustDead) in `HEMCO_Config.rc`. Then:
- We may not need to define `METDIR:` in `HEMCO_Config.rc` because it has not been used therein;
- Will `Mass tuning factor     :` only differ between 4x5 and 2x2.5 simulations? Or the higher the horizontal resolutions, the lower the values? Users should be reminded if they create nested run directories based on their global run directories.
```
<     --> Mass tuning factor     :       7.8533e-4
---
>     --> Mass tuning factor     :       4.7586e-4
```

I have started running these several options and `GC.log` files will be attached here soon. Stay tuned!

**Comment by FeiYao-Edinburgh** [comment]
Hello again. Unfortunately, I have got the same error as before for options 2 to 5, which suggests that this wet deposition issue is indeed related to fullchem regardless of either horizontal or vertical resolutions. For option 6, while `gcclassic` can pass "2019/01/05 00:40" when the wet deposition issue occurred since we turn of wet deposition, it immediately dies at "2019/01/05 00:50" in `DO_CLOUD_CONVECTION`. As such I do not think I have identified the root of evil backwards. All `GC.log` files are attached below for your reference.

[GC_original.log](https://github.com/geoschem/geos-chem/files/10929865/GC_original.log)
[GC_option_2.log](https://github.com/geoschem/geos-chem/files/10929866/GC_option_2.log)
[GC_option_3.log](https://github.com/geoschem/geos-chem/files/10929867/GC_option_3.log)
[GC_option_4.log](https://github.com/geoschem/geos-chem/files/10929861/GC_option_4.log)
[GC_option_5.log](https://github.com/geoschem/geos-chem/files/10929863/GC_option_5.log)
[GC_option_6.log](https://github.com/geoschem/geos-chem/files/10929864/GC_option_6.log)

I recall that I have turned off grid-independent offline emissions to use their online alternatives instead. Not sure if that's the cause but I will roolback such practice to see if helps:
- Option 7: rollback to use the several OFFLINE emissions

Then I may use the gdb debugger trying to debug in depth. But at the same time, can you help reproduce and solve the issue at your earliest convenience? Thanks!

All the best,
Fei

**Comment by FeiYao-Edinburgh** [comment]
I finally found where the error comes from via Option 7! Personally, I'd like to stick to online emissions so as to save space. However, if I modify the following items in `HEMCO_Config.rc` my simulations restarted at 2019/01/01 will crash at 2019/01/05T00:40 and prompts the wet deposition issue as described above.
```
    --> OFFLINE_DUST           :       true     # 1980-2019
    --> OFFLINE_BIOGENICVOC    :       true     # 1980-2020
    --> OFFLINE_SEASALT        :       true     # 1980-2019
    -->  CalcBrSeasalt         :       false
    --> OFFLINE_SOILNOX        :       true     # 1980-2020
```
=>
```
    --> OFFLINE_DUST           :       false    # 1980-2019
    --> OFFLINE_BIOGENICVOC    :       false    # 1980-2020
    --> OFFLINE_SEASALT        :       false    # 1980-2019
    -->  CalcBrSeasalt         :       false
    --> OFFLINE_SOILNOX        :       false    # 1980-2020
```
And
```
104     SoilNOx                : off   NO
105     DustDead               : off   DST1/DST2/DST3/DST4
107     SeaSalt                : off   SALA/SALC/SALACL/SALCCL/SALAAL/SALCAL/BrSALA/BrSALC/MOPO/MOPI
```
=>
```
104     SoilNOx                : on    NO
105     DustDead               : on    DST1/DST2/DST3/DST4
107     SeaSalt                : on    SALA/SALC/SALACL/SALCCL/SALAAL/SALCAL/BrSALA/BrSALC/MOPO/MOPI
```
Btw, `108 MEGAN` is always on so I do not need to change it. I think that's all I need to do if I want to use online rather than offline emissions according to this [guide](http://wiki.seas.harvard.edu/geos-chem/index.php/Grid-independent_emissions).

In that guide I also see the dependence of `Mass tuning factor     :` on the horizontal resolutions. I do not think that has been informed to users if they want to create their nested run directories based on their global run directories themselves in this [page](http://wiki.seas.harvard.edu/geos-chem/index.php/Setting_up_GEOS-Chem_nested_grid_simulations#GEOS-Chem_12.4.0_and_later). Meanwhile, users should be reminded that they need to turn off the `BoundaryConditions` collection in `HISTORY.rc` when they run nested simulations.

Therefore, this bug is essentially associated with HEMCO. While I can get rid the issue by using the offline emissions, I'd be graetful if you could make the online emissions work again so we may save our space.

All the best,
Fei

**Comment by FeiYao-Edinburgh** [comment]
Hello again. Unfortunately, while a rollback to offline emissions can get pass 2019/01/05, it cannot finished the entire month. The model crashes again at 2019/01/18:
```
===============================================================================
WETDEP: ERROR at  120  65   1 for species    2 in area WASHOUT: at surface
 LS          :  T
 PDOWN       :    0.0000000000000000     
 QQ          :    0.0000000000000000     
 ALPHA       :    0.0000000000000000     
 ALPHA2      :    0.0000000000000000     
 RAINFRAC    :    0.0000000000000000     
 WASHFRAC    :    0.0000000000000000     
 MASS_WASH   :    0.0000000000000000     
 MASS_NOWASH :    0.0000000000000000     
 WETLOSS     :                        NaN
 GAINED      :    0.0000000000000000     
 LOST        :    0.0000000000000000     
 DSpc(NW,:)  :                        NaN   6.7205742246851682E-013   7.5368436847474845E-013   8.4975796089099128E-013   9.5924340586099975E-013   1.0829924984759400E-012   1.2258502009621605E-012   1.3901956656657957E-012   1.5680142299667902E-012   1.7551515945906939E-012   1.9547159549426595E-012   2.1651635090475925E-012   2.3385150704061026E-012   2.4554694949496124E-012   2.3357668657715943E-012   1.7072805381903821E-012   1.7072805381903821E-012   1.7072805381903821E-012   1.7072805381903821E-012   1.7072805381903821E-012   1.7072545155447299E-012   1.7055418510464183E-012   1.3927881414221677E-012   4.4003755616397899E-013   0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000        0.0000000000000000     
 Spc(I,J,:N) :                        NaN   1.9291128559612601E-008   2.0167723640550359E-008   2.0274707672473212E-008   2.0738585195427509E-008   2.0769596092871515E-008   2.0145038161660465E-008   1.8730796314899240E-008   1.7030707068122193E-008   1.5006053553852801E-008   1.3533668172022698E-008   1.3010651203617914E-008   1.7386194767552557E-008   2.1875654200427112E-008   2.1832284847411572E-008   2.1820496969362129E-008   2.1132044546042722E-008   2.1398265581855485E-008   3.5284881019345504E-008   3.6970559337065084E-008   3.5443998480763681E-008   3.2121319094602886E-008   2.7120446807596470E-008   2.1945816973181532E-008   2.0759824582379950E-008   2.2137879126277988E-008   2.1218184059487257E-008   2.1444214874976738E-008   9.9282979947711402E-009   3.8535753429663224E-009   2.0743543378641662E-009   7.4445289954501193E-010   1.6393842510846428E-010   5.8298789751468229E-011   2.6745082751238423E-011   1.2767078273196251E-011   8.2706045672440057E-012   9.9594526015373138E-013   5.8125721694965122E-014   2.0090850758815558E-015   2.0952854652581820E-016   3.0544729517472366E-015   5.9320574128330520E-015   1.2361747938570271E-013   2.2203141257719241E-012   4.6034785345569593E-012   4.7659789609676989E-012
===============================================================================
===============================================================================
GEOS-Chem ERROR: Error encountered in wet deposition!
 -> at SAFETY (in module GeosCore/wetscav_mod.F90)
===============================================================================

===============================================================================
GEOS-Chem ERROR: Error encountered in "Safety"!
 -> at Do_Washout_at_Sfc (in module GeosCore/wetscav_mod.F90)
===============================================================================

===============================================================================
GEOS-Chem ERROR:
 -> at WetDep (in module GeosCore/wetscav_mod.F90)
===============================================================================

===============================================================================
GEOS-Chem ERROR: Error encountered in "Wetdep"!
 -> at Do_WetDep (in module GeosCore/wetscav_mod.F90)
===============================================================================

===============================================================================
GEOS-CHEM ERROR: Error encountered in "Do_WetDep"!
STOP at  -> at GEOS-Chem (in GeosCore/main.F90)
===============================================================================
     - CLEANUP: deallocating arrays now...

-------------------------------------------------------------------------------
HEMCO 3.6.2 FINISHED.
Warnings (level 1 or lower):   1066
-------------------------------------------------------------------------------
```

So I suppose we may really cannot ignore this issue.

**Comment by lizziel** [comment]
Hi @FeiYao-Edinburgh, I have some other priorities going on right now but will get to this when I can to see if I can reproduce.

**Comment by lizziel** [comment]
Hi @FeiYao-Edinburgh, I was able to reproduce this. I'll let you know when I have the solution. See also https://github.com/geoschem/GCHP/issues/302.

**Comment by lizziel** [comment]
Hi @FeiYao-Edinburgh, please see a fix [here](https://github.com/geoschem/geos-chem/pull/1698). Let us know if this does not solve the problem.

**Comment by FeiYao-Edinburgh** [comment]
Thanks @lizziel I can confirm that the issue has gone!

**Comment by lizziel** [comment]
Excellent! If you have any other questions (I think you had a few other unrelated questions/ideas in some of your comments), please make a new issue to discuss.
